source/server/server_configuration_parser.cpp

```diff
@@ -287,6 +292,47 @@ int ServerConfigurationParser::parse_port_with_key(const std::string& key) const
   return parsed_port;
 }
 
+CpuAffinityConfiguration ServerConfigurationParser::parse_cpu_affinity() const
+{
+    CpuAffinityConfiguration cpu_affinity;
+
+    auto core_config_it = config_file_.find(kCpuAffinityConfigurationKey);
+    if (core_config_it != config_file_.end()) {
+        cpu_affinity.sideband_read_write = parse_cpu_affinity_with_key(kSidebandReadWriteKey);
+        cpu_affinity.stream_write = parse_cpu_affinity_with_key(kStreamWriteKey);
+        cpu_affinity.server = parse_cpu_affinity_with_key(kServerKey);
+    }
+    else{
+      // -1 is set as the default value for the cores which indicates that any available core can be used.
+      cpu_affinity.sideband_read_write = -1;
+      cpu_affinity.stream_write = -1;
+      cpu_affinity.server = -1;
+    }
```

Comment:
I feel like the struct should be set to the default values on creation. That way it is in a valid state even if one of the parse function throws an exception.

Reply:
That makes sense, set it to default value in while creating in `cpu_affinity_configuration.h` and removed the else block highlighted above.